packages/front-end/components/Metrics/MetricForm.tsx

```diff
@@ -321,7 +327,11 @@ const MetricForm: FC<MetricFormProps> = ({
       userIdType: value.userIdTypes.join(", "),
     });
 
-    onClose(true);
+    if (cta === "Next: Create Experiment") {
+      onSuccess();
+    } else {
+      onClose(true);
+    }
```

Comment:
Let's change this component to just have a `onClose` prop that doesn't take any arguments.  Then we can rely on the built-in behavior that auto-closes on submit.  This logic then becomes much simpler:

```tsx
onSuccess && onSuccess();
```